src/components/JourneyComparison/JourneyComparison.tsx

```diff
@@ -33,26 +39,27 @@ const JourneyComparison = (props: ComparisonProps) => {
 
     const baselineTotalDuration = getTotalJourneyDuration(baseline);
     const enhancedTotalDuration = getTotalJourneyDuration(enhanced);
-    const enhancedTotalFraction = 1 - enhancedTotalDuration / baselineTotalDuration;
+    const enhancedTotalFraction =
+        baselineTotalDuration > 0 ? 1 - enhancedTotalDuration / baselineTotalDuration : 0;
     const showDelayRow =
-        enhanced.amenities.includes("electricTrains") &&
-        !baseline.amenities.includes("electricTrains");
-    const showWaitRow = !baseline.reverse;
-    const amenitiesDiff = enhanced.amenities.filter((a) => !baseline.amenities.includes(a));
+        enhanced?.amenities?.includes("electricTrains") &&
```

Comment:
With a fallback `JourneyInfo` object we won't need these `?.` operators since all of these fields will still be here!